[DO NOT MERGE] Add support for T-SQL INFORMATION_SCHEMA.SEQUENCES

aadityavermaa opened the pull request:
### Description
In this commit, created a new view information_schema_tsql.sequences view to support T-SQL INFORMATION_SCHEMA.SEQUENCES

Signed-off-by: Aditya Verma <adiityav@amazon.com>

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

Commit:
BABEL-3452:Create View INFORMATION_SCHEMA.SEQUENCES

Creation of information_schema_tsql.sequences view to support SQL
Server's INFORMATION_SCHEMA.SEQUENCES

Signed-off-by: Aditya Verma <adiityav@amazon.com>

KushaalShroff commented on contrib/babelfishpg_tsql/sql/information_schema_tsql.sql:
```diff
@@ -716,4 +716,36 @@ CREATE OR REPLACE VIEW information_schema_tsql.routines AS
+        WHERE nc.oid = r.relnamespace
+            AND r.relkind = 'S'
+            AND (NOT pg_is_other_temp_schema(nc.oid))
+            AND (pg_has_role(r.relowner, 'USAGE')
```
Add another condition for 
ext.dbid = cast(sys.db_id() as oid) which applies to cross db, I think this was working because the roles in different databases do not have privilege  to that object.

KushaalShroff commented on contrib/babelfishpg_tsql/sql/information_schema_tsql.sql:
```diff
@@ -716,4 +716,36 @@ CREATE OR REPLACE VIEW information_schema_tsql.routines AS
+            CAST(s.seqmax AS int) AS "MAXIMUM_VALUE",
+            CAST(s.seqincrement  AS int) AS "INCREMENT",
+            CAST( CASE WHEN s.seqcycle = 't' THEN 1 ELSE 0 END AS int) AS "CYCLE_OPTION",
+            CAST(NULL AS sys.nvarchar(128)) AS "DECLARED_DATA_TYPE",
```
Add a comment that this needs to be supported after the bug for sequences is fixed

KushaalShroff commented on contrib/babelfishpg_tsql/sql/information_schema_tsql.sql:
```diff
@@ -716,4 +716,36 @@ CREATE OR REPLACE VIEW information_schema_tsql.routines AS
+    SELECT CAST(nc.dbname AS sys.nvarchar(128)) AS "SEQUENCE_CATALOG",
+            CAST(extc.orig_name AS sys.nvarchar(128)) AS "SEQUENCE_SCHEMA",
+            CAST(r.relname AS sys.nvarchar(128)) AS "SEQUENCE_NAME",
+            CAST(CASE WHEN tsql_type_name = 'sysname' THEN sys.translate_pg_type_to_tsql(t.typbasetype) ELSE tsql_type_name END
```
Add a comment that for numeric and decimal we display bigint datatype which is due Postgres' inherent implementation.

Commit:
BABEL-3452: Support T-SQL INFORMATION_SCHEMA.SEQUENCES View

Creation of view information_schema_tsql.sequences to support for SQL server's INFORMATION_SCHEMA.SEQUENCES.

Added comments and fixed a bug
Also added upgrade testcases
Signed-off-by: Aditya Verma <adiityav@amazon.com>
